signing_service/src/signing_service.py

```diff
@@ -0,0 +1,151 @@
+import argparse
+import logging.config
+import os
+import signal
+import socket
+from time import sleep
+
+from constants import SIGNING_SERVICE_DEFAULT_PORT
+from constants import SIGNING_SERVICE_MAXIMUM_RECONNECT_TIME
+
+
+logger = logging.getLogger()
+crash_logger = logging.getLogger('crash')
+
+
+class SigningService:
+    """
+    The Signing Service connects to Middleman as a client but then listens for requests coming from Concent via
+    Middleman. The underlying protocol is TCP and data sent over that is expected to conform to the Wire protocol.
+    """
+
+    def __init__(self, host, port, reconnect_delay) -> None:
+        assert isinstance(host, str)
+        assert isinstance(port, int)
+        assert isinstance(reconnect_delay, int)
+        self.host = host
+        self.port = port
+        self.reconnect_delay = reconnect_delay
+        self.current_reconnect_delay = 0
```

Comment:
I think it might be a good idea to use `__slots__`. It lets you declare fields ahead of time and also has smaller memory footprint.

Reply:
:+1: done